### PR TITLE
new API LeaderChangeEvent

### PR DESCRIPTION
**Pull request by MertUnverdi**
our api is not rich, we could add more events :+1:

### CODE REVIEW

**Review by dlsf — CHANGES_REQUESTED** [review]
In my opinion the event should either be triggered after the confirmation or be changed to something like PreLeaderChangeEvent

**Comment by dlsf** [comment]
I don't know if this is a useful event tho, I guess we have to wait for @PeachesMLG. An event triggered when the leader actually changes would probably be more useful.

**Comment by dlsf** [comment]
Also, this event should be triggered by the MembersCommand and MembersGUI

**Comment by PeachesMLG** [comment]
I dont see why this is needed tbh, why not have it after the confirmation gui?

**Comment by MertUnverdi** [comment]
cause someone wants ignoring leader change

**Comment by dlsf** [comment]
But he can still do that when the event gets triggered after the confirmation before the leader actually changes

**Comment by MertUnverdi** [comment]
nope if event is not cancelled
event will work after then transfer finished
is wrong??
im still thinking is should be before then event cause someones can be wants cancel transfer without permission or anything
ik this plugin doesn't supports p2w but this not counts tho :D

**Comment by MertUnverdi** [comment]
but if u say change it to before, i can np <3

**Review by dlsf — APPROVED** [review]

File: src/main/java/com/iridium/iridiumskyblock/api/PreLeaderChangeEvent.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.iridium.iridiumskyblock.api;
+
+import com.iridium.iridiumskyblock.Island;
+import com.iridium.iridiumskyblock.User;
+import org.bukkit.event.Cancellable;
+import org.jetbrains.annotations.NotNull;
+
+public class PreLeaderChangeEvent extends IslandEvent implements Cancellable {
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/api/PreLeaderChangeEvent.java:8]: Rename the event again

File: src/main/java/com/iridium/iridiumskyblock/commands/TransferCommand.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import com.iridium.iridiumskyblock.IridiumSkyblock;
 import com.iridium.iridiumskyblock.Island;
 import com.iridium.iridiumskyblock.User;
+import com.iridium.iridiumskyblock.api.PreLeaderChangeEvent;
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/commands/TransferCommand.java:6]: Remove this import

**Review by dlsf — CHANGES_REQUESTED** [review]

**Pull request merged**